crates/loro-internal/src/container/richtext/richtext_state.rs

```diff
@@ -1868,6 +1922,54 @@ impl RichtextState {
         Ok(ans)
     }
 
+    pub(crate) fn get_text_slice_by_event_index(
+        &self,
+        pos: usize,
+        len: usize,
+    ) -> LoroResult<String> {
+        if self.tree.is_empty() {
+            return Ok(String::new());
+        }
+
+        if len == 0 {
+            return Ok(String::new());
+        }
+
+        if pos + len > self.len_event() {
+            return Err(LoroError::OutOfBound {
+                pos: pos + len,
+                len: self.len_event(),
+                info: format!("Position: {}:{}", file!(), line!()).into_boxed_str(),
+            });
+        }
+
+        let mut ans = String::new();
+        let (start, end) = (
+            self.tree.query::<EventIndexQuery>(&pos).unwrap().cursor,
+            self.tree
+                .query::<EventIndexQuery>(&(pos + len))
+                .unwrap()
+                .cursor,
+        );
+
+        for span in self.tree.iter_range(start..end) {
+            let start = span.start.unwrap_or(0);
+            let end = span.end.unwrap_or(span.elem.rle_len());
+            if end == 0 {
+                break;
+            }
+
+            if let RichtextStateChunk::Text(s) = span.elem {
+                match slice_string_by_event_index(&s.as_str(), start, end) {
```

Comment:
We should use `slice_string_by_unicode_index` here. 

There should be a test case that can make this fail on the JS side